src/utils/job/tests/test_task.py

```
@@ -931,5 +931,520 @@ def test_mixed_credentials(self):
+    """Tests for TaskInputOutput regex validation and workflow info parsing."""
+
+    def test_empty_regex_passes(self):
+        spec = task.TaskInputOutput(task='task1', regex='')
+        self.assertEqual(spec.regex, '')
+
+    def test_valid_regex_passes(self):
+        spec = task.TaskInputOutput(task='task1', regex=r'.*\.txt$')
+        self.assertEqual(spec.regex, r'.*\.txt$')
+
+    def test_invalid_regex_raises(self):
+        with self.assertRaises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)